src/lib/deploy.js

```diff
@@ -539,14 +553,14 @@ When o1js and the zkApp CLI have a major version of 1 or higher,
 changes to the major version of the zkApp CLI will represent
 breaking changes, following semver.
 */
-function hasBreakingChanges(version1, version2) {
-  const version1Arr = version1?.split('.');
-  const version2Arr = version2?.split('.');
+function hasBreakingChanges(installedVersion, latestVersion) {
+  const installedVersionArr = installedVersion?.split('.');
```

Comment:
nit: You could do `installedVersion?.split('.').map(Number)` if you would like to avoid casting twice in the return statements